vortex/src/main/java/org/apache/iceberg/vortex/VortexIterable.java

```diff
@@ -23,214 +23,225 @@
-            qualified = "https://" + qualified;
-          }
-          properties.setEndpoint(qualified);
-          break;
-        case FS_S3A_ENDPOINT_REGION:
-          properties.setRegion(entry.getValue());
-          break;
-        default:
-          LOG.trace(
-              "Ignoring unknown s3a connector property: {}={}", entry.getKey(), entry.getValue());
-          break;
-      }
-    }
-
-    return properties.asProperties();
-  }
-
-  static class VortexBatchIterator implements CloseableIterator<Array> {
-    private PrefetchingIterator<Array> stream;
-
-    private VortexBatchIterator(PrefetchingIterator<Array> stream) {
-      this.stream = stream;
+    private static final Logger LOG = LoggerFactory.getLogger(VortexIterable.class);
+
+    private final InputFile inputFile;
+    private final Optional<Expression> filterPredicate;
+    private final Optional<DeleteFilter<?>> deleteFilter;
+    private final Function<DType, VortexRowReader<T>> rowReaderFunc;
+    private final Function<DType, VortexBatchReader<T>> batchReaderFunction;
+    private final List<String> projection;
+
+    VortexIterable(
+            InputFile inputFile,
+            Schema icebergSchema,
+            Optional<Expression> filterPredicate,
+            Optional<DeleteFilter<?>> deleteFilter,
+            Function<DType, VortexRowReader<T>> readerFunction,
+            Function<DType, VortexBatchReader<T>> batchReaderFunction) {
+        this.inputFile = inputFile;
+        // We have the file schema, we need to assign Iceberg IDs to the entire file schema
+        this.projection = Lists.transform(icebergSchema.columns(), Types.NestedField::name);
+        this.filterPredicate = filterPredicate;
+        this.deleteFilter = deleteFilter;
+        this.rowReaderFunc = readerFunction;
+        this.batchReaderFunction = batchReaderFunction;
     }
 
     @Override
-    public Array next() {
-      return stream.next();
+    public CloseableIterator<T> iterator() {
+        File vortexFile = newVortexFile(inputFile);
+        addCloseable(vortexFile);
+
+        // Return the filtered scan, and then the projection, etc.
+        Optional<dev.vortex.api.Expression> scanPredicate =
+                filterPredicate.map(
+                        icebergExpression -> {
+                            Schema fileSchema = VortexSchemas.convert(vortexFile.getDType());
+                            return ConvertFilterToVortex.convert(fileSchema, icebergExpression);
+                        });
+
+        Optional<SerializedBitmap> serializedBitmap = deleteFilter.flatMap(DeleteFilter::bitmapBytes)
+                .map(bitmapBytes -> ImmutableSerializedBitmap.of(bitmapBytes.getBytes(), bitmapBytes.getOffset(), bitmapBytes.getLength(), false));
+
+        ArrayStream batchStream =
+                vortexFile.newScan(
+                        ScanOptions.builder().addAllColumns(projection).predicate(scanPredicate).selectionBitmap(serializedBitmap).build());
```

Comment:
this is the related change in this file. Might wanna review this with ignore whitespace set. I think my IDE reformatted this